root/etc/e-smith/events/actions/restore-config-download

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+#
+# Copyright (C) 2019 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+packagelist_path=/var/lib/nethserver/backup/package-list
+backup_path=/var/lib/nethserver/backup/backup-config.tar.xz
+
+packages=($(tar -x -O -f ${backup_path} ${packagelist_path:1} | sed -f /etc/nethserver/ns6upgrade-package-list.sed))
+
+if (( ${#packages[@]} == 0 )); then
+    echo "[WARNING] no packages listed for download"
```

Comment:
I think this shouldn't be a WARNING but just a notice

Reply:
I expect the list always contains some elements. If it is empty it is an abnormal condition and I want to raise it, though I can't consider it an error.